File: Python/Article.py
from ArticleState import ArticleState

# Originator


class Article:
    def __init__(self, title, content):
        self.__title = title
        self.__content = content

    def create_state(self):
        return ArticleState(self.__title, self.__content)

    def restore(self, state):
        if state:
            self.__title = state.title
            self.__content = state.content

    @property
    def title(self):
        return self.__title

    @title.setter
    def title(self, value):
        self.__title = value

    @property
    def content(self):
        return self.__content

    @content.setter
    def content(self, value):
        self.__content = value

    def __str__(self):
        return f"Title: {self.__title}\nContent: {self.__content}"

File: JS/Article.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const ArticleState_1 = __importDefault(require("./ArticleState"));
// Originator
class Article {
    constructor(title, content) {
        this.title = title;
        this.content = content;
    }
    createState() {
        return new ArticleState_1.default(this.title, this.content);
    }
    restore(state) {
        if (state) {
            this.title = state.getTitle();
            this.content = state.getContent();
        }
    }
    getTitle() {
        return this.title;
    }
    setTitle(title) {
        this.title = title;
    }
    getContent() {
        return this.content;
    }
    setContent(content) {
        this.content = content;
    }
    toString() {
        return `Title: ${this.title}\nContent: ${this.content}`;
    }
}
exports.default = Article;
